src/Microsoft.TemplateEngine.Cli/New3Command.cs

```diff
@@ -97,13 +96,47 @@ public New3Command(string commandName, ITemplateEngineHost host, Action<IEngineE
 
         public bool SkipUpdateCheck => _app.InternalParamHasValue("--skip-update-check");
 
-        public IFilteredTemplateInfo UnambiguousTemplateToUse
+        private static bool AreAllTemplatesSameGroupIdentity(IEnumerable<IFilteredTemplateInfo> templateList)
+        {
+            Func<IFilteredTemplateInfo, string> groupIdentitySelector = (x) => x.Info.GroupIdentity;
+            return templateList.AllAreTheSame(groupIdentitySelector, StringComparer.OrdinalIgnoreCase);
```

Comment:
Seems more readable to me, but I don't feel particularly strongly about it either way.